lib/Echidna/Exec.hs

```diff
@@ -69,25 +70,30 @@ vmExcept e = throwM $ case VMFailure e of {Illegal -> IllegalExec e; _ -> Unknow
 -- using the given execution strategy, handling errors with the given handler.
 execTxWith :: (MonadState x m, Has VM x) => (Error -> m ()) -> m VMResult -> Tx -> m (VMResult, Int)
```

Comment:
is there any reason why we can't just keep this to the gas type to avoid `fromIntegral` at the end?